src/icp_canister_backend/tests/test.rs

```diff
@@ -273,17 +303,8 @@ fn test_successful_withdrawal() {
         .expect("Failed to check user initial balance");
     let initial_balance: Nat = decode_one(&initial_balance_result).unwrap();
 
-    create_deposit(
-        &pic,
-        canister_id,
-        ledger_id,
-        user,
-        deposit_amount.clone(),
-        timelock,
-    );
-
-    pic.advance_time(std::time::Duration::from_secs(timelock + 1));
-    pic.tick();
+    // Create deposit and advance time to simulate finished episode
+    create_deposit_and_advance_time(&pic, canister_id, ledger_id, user, deposit_amount.clone());
```

Comment:
Don't make such a merged functionality function.
Keep one function create_deposit.
Create another one advance_time with time variable to increase.